array_api_strict/_array_object.py

```diff
@@ -625,19 +661,21 @@ def __getitem__(
         """
         Performs the operation __getitem__.
         """
+        # XXX Does key have to be on the same device? Is there an exception for CPU_DEVICE?
```

Comment:
I guess this is underspecified. I would need to test what PyTorch and others do, but I would suspect that an implicit cross-device array `key` is not something that's intended to be supported, since that still would require an implicit device transfer.